pkgs/development/libraries/nlopt/default.nix

```diff
@@ -2,50 +2,149 @@
   lib,
   stdenv,
   fetchFromGitHub,
+  fetchpatch,
   cmake,
-  octave ? null,
-  libiconv,
-}:
+  nix-update-script,
+  # Optionally build Python bindings
+  withPython ? false,
+  python3,
+  python3Packages,
+  # Optionally build Octave bindings
+  withOctave ? false,
+  octave,
+  # Optionally build Java bindings
+  withJava ? false,
+  jdk,
+  # Required for building the Python and Java bindings
+  swig,
+  # Optionally exclude Luksan solvers to allow licensing under MIT
+  withoutLuksanSolvers ? false,
+  # Build static on-demand
+  withStatic ? stdenv.hostPlatform.isStatic,
 
-stdenv.mkDerivation rec {
+  # v2.8.0 introduced a regression where testing on Linux platforms fails with a buffer overflow
+  # when compiled with -D_FORTIFY_SOURCE=3.
+  # This was deemed to be a compiler false positive by the library's author in https://github.com/stevengj/nlopt/issues/563.
```

Comment:
The linked issue references a [fix included in 2.10.0](https://github.com/stevengj/nlopt/commit/487b87fc428c7c9c32c1cf1150b2dcc4e4d446f5), so is this still needed?

Reply:
The regression first emerged when compiling 2.10.0 for nixpkgs, so it is still needed. At the bottom, there is someone who apparently could reproduce it, so it seems we're not alone.